### PR TITLE
{Release} Upgrade to Azure CLI 2.52.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

**Comment by yonzhan** [comment]
Release

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,103 @@
...
+* `az aks create/update`: Add new parameter `--enable-vpa` to support enabling vertical pod autoscaler for cluster (#27019)
+* `az aks update`: Add new parameter `--network-dataplane` to specify the network dataplane used in the Kubernetes cluster (#27060)
+* `az aks create/update`: Add new parameter `--node-os-upgrade-channel` to specify which OS on your nodes is updated (#27167)
+* `az aks update`: Retain value in network profile in mc object only when decorator is in update mode (#27050)
+* `az aks update`: Outbound ip/ outbound ipprefix and managed outbound ip should be mutually exclusive (#27271)
+
+**App Config**
+
+* `az apponfig kv import`: Add new parameter `--import-mode` to specify whether to overwrite already existing key-values or ignore matching keys (#26098)
+* `az appconfig kv export`: Add a new parameter `--snapshot` to support exporting all key values from a snapshot of the source configuration (#27043)
```
Comment (wangzelin007) [inline src/azure-cli/HISTORY.rst:20]:
```suggestion
* `az appconfig kv export`: Add new parameter `--snapshot` to support exporting all key values from a snapshot of the source configuration (#27043)
```

Comment (wangzelin007) [inline src/azure-cli/HISTORY.rst:20]:
```suggestion
* `az appconfig kv export`: Add new parameter `--snapshot` to support exporting all key values from a snapshot of the source configuration (#27043)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,103 @@
...
+* Fix #26112: `az deployment group create`: Fix the warning log `mode is not a known attribute of class TemplateLink` (#26984)
+* `az bicep build-params`: Support generating `parameters.json` file from the given `bicepparam` file with the `--file` argument (#26781)
+* `az bicep decompile-params`: Support generating `parameters.bicepparam` file from the given `parameters.json` file with the `--file` argument (#26781)
+* `az bicep generate-params`: Support generating `main.parameters.json` with the parameters that doesn't have default values in the given `.bicep` file (#26781)
+* `az bicep generate-params`: Add new parameter `--output-format` to support generating parameter file in `bicepparam` and `json` formats (#26781)
+* `az bicep generate-params`: Add new parameter `--include-params` to support generating parameter file with all the parameters in the given `bicep` file, or with only parameters that doesn't have default values in the given `bicep` file (#26781)
+
+**ARO**
+
+* `az aro` (all commands): Update to v20230401 API (#27212)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:39]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,103 @@
...
+* `az aks update`: Add new parameter `--network-dataplane` to specify the network dataplane used in the Kubernetes cluster (#27060)
+* `az aks create/update`: Add new parameter `--node-os-upgrade-channel` to specify which OS on your nodes is updated (#27167)
+* `az aks update`: Retain value in network profile in mc object only when decorator is in update mode (#27050)
+* `az aks update`: Outbound ip/ outbound ipprefix and managed outbound ip should be mutually exclusive (#27271)
+
+**App Config**
+
+* `az apponfig kv import`: Add new parameter `--import-mode` to specify whether to overwrite already existing key-values or ignore matching keys (#26098)
+* `az appconfig kv export`: Add a new parameter `--snapshot` to support exporting all key values from a snapshot of the source configuration (#27043)
+* `az appconfig kv import`: Add a new parameter `--src-snapshot` to support importing all key values from a snapshot of the source configuration (#27043)
```
Comment (wangzelin007) [inline src/azure-cli/HISTORY.rst:21]:
```suggestion
* `az appconfig kv import`: Add new parameter `--src-snapshot` to support importing all key values from a snapshot of the source configuration (#27043)
```

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:39]: API version bump is not customer-facing change. BTW, `v20230401` is not a valid API version format. 

```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,103 @@
...
+2.52.0
+++++++
+
+**AKS**
+
+* `az aks create/update`: Add new parameter `--enable-vpa` to support enabling vertical pod autoscaler for cluster (#27019)
+* `az aks update`: Add new parameter `--network-dataplane` to specify the network dataplane used in the Kubernetes cluster (#27060)
+* `az aks create/update`: Add new parameter `--node-os-upgrade-channel` to specify which OS on your nodes is updated (#27167)
+* `az aks update`: Retain value in network profile in mc object only when decorator is in update mode (#27050)
+* `az aks update`: Outbound ip/ outbound ipprefix and managed outbound ip should be mutually exclusive (#27271)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:15]:
```suggestion
* `az aks update`: Outbound ip/outbound ipprefix and managed outbound ip should be mutually exclusive (#27271)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,103 @@
...
+**Redis**
+
+* `az redis update`: Fix public network access default value issue (#27227)
+
+**Storage**
+
+* [BREAKING CHANGE] `az storage account create`: Server change default value for `--allow-blob-public-access` and `--allow-cross-tenant-replication` to False for security concerns (#27048)
+* `az storage file upload-batch`: Allow uploading files in parallel to improve performance (#26940)
+* `az storage account create/update`: Revert `--allow-cross-tenant-replication` and `--allow-blob-public-access` to default value of true (#27250)
+* Fix #27202 `az storage entity insert`: Fix case when using sas token with only `add` permission (#27280)
```
Comment (bebound) [inline src/azure-cli/HISTORY.rst:97]:
```suggestion
* Fix #27202: `az storage entity insert`: Fix case when using sas token with only `add` permission (#27280)
```

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (calvinhzy) [inline src/azure-cli/HISTORY.rst:39]:
```suggestion
* `az aro validate`: Add validate command (#27212)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,103 @@
...
+
+**Upgrade**
+
+* `az upgrade`: Change MSI URL based on the installed CLI architecture (#27104)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:101]:
```suggestion
* `az upgrade`: Support upgrading 64-bit MSI (#27104)
```

Comment (jiasli) [reply src/azure-cli/HISTORY.rst:101]:
```suggestion
* `az upgrade`: Support upgrading with 64-bit MSI (#27104)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,103 @@
...
+* `az network application-gateway waf-policy managed-rule rule-set`: Support Microsoft_BotManagerRuleSet version 1.0 (#27184)
+* `az network vnet peering create`: Mark `--remote-vnet` as required (#27198)
+
+**Redis**
+
+* `az redis update`: Fix public network access default value issue (#27227)
+
+**Storage**
+
+* [BREAKING CHANGE] `az storage account create`: Server change default value for `--allow-blob-public-access` and `--allow-cross-tenant-replication` to False for security concerns (#27048)
```
Comment (calvinhzy) [inline src/azure-cli/HISTORY.rst:94]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,103 @@
...
+* `az storage file upload-batch`: Allow uploading files in parallel to improve performance (#26940)
+* `az storage account create/update`: Revert `--allow-cross-tenant-replication` and `--allow-blob-public-access` to default value of true (#27250)
```
Comment (calvinhzy) [inline src/azure-cli/HISTORY.rst:96]:
```suggestion
```

**Review by evelyn-ys — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Pull request merged**